cns/types/codes.go

```diff
@@ -3,47 +3,53 @@ package types
 type ResponseCode int
 
 // ResponseCode definitions
+// 4000 codes cannot be retried
+// 5000 codes can be retried
+// x100 codes are related to network, connectivity, and environment
+// x300 codes are related to container, orchestrator, and virtualization
+// x500 codes are related to authorization and unexpected errors
+// x700 codes are related agent and internal server errors
 const (
-	Success                                ResponseCode = 0
-	UnsupportedNetworkType                 ResponseCode = 1
-	InvalidParameter                       ResponseCode = 2
-	UnsupportedEnvironment                 ResponseCode = 3
-	UnreachableHost                        ResponseCode = 4
-	ReservationNotFound                    ResponseCode = 5
-	MalformedSubnet                        ResponseCode = 8
-	UnreachableDockerDaemon                ResponseCode = 9
-	UnspecifiedNetworkName                 ResponseCode = 10
-	NotFound                               ResponseCode = 14
-	AddressUnavailable                     ResponseCode = 15
-	NetworkContainerNotSpecified           ResponseCode = 16
-	CallToHostFailed                       ResponseCode = 17
-	UnknownContainerID                     ResponseCode = 18
-	UnsupportedOrchestratorType            ResponseCode = 19
-	DockerContainerNotSpecified            ResponseCode = 20
-	UnsupportedVerb                        ResponseCode = 21
-	UnsupportedNetworkContainerType        ResponseCode = 22
-	InvalidRequest                         ResponseCode = 23
-	NetworkJoinFailed                      ResponseCode = 24
-	NetworkContainerPublishFailed          ResponseCode = 25
-	NetworkContainerUnpublishFailed        ResponseCode = 26
-	InvalidPrimaryIPConfig                 ResponseCode = 27
-	PrimaryCANotSame                       ResponseCode = 28
-	InconsistentIPConfigState              ResponseCode = 29
-	InvalidSecondaryIPConfig               ResponseCode = 30
-	NetworkContainerVfpProgramPending      ResponseCode = 31
-	FailedToAllocateIPConfig               ResponseCode = 32
-	EmptyOrchestratorContext               ResponseCode = 33
-	UnsupportedOrchestratorContext         ResponseCode = 34
-	NetworkContainerVfpProgramComplete     ResponseCode = 35
-	NetworkContainerVfpProgramCheckSkipped ResponseCode = 36
-	NmAgentSupportedApisError              ResponseCode = 37
-	UnsupportedNCVersion                   ResponseCode = 38
-	FailedToRunIPTableCmd                  ResponseCode = 39
-	NilEndpointStateStore                  ResponseCode = 40
-	NmAgentInternalServerError             ResponseCode = 41
-	StatusUnauthorized                     ResponseCode = 42
-	UnsupportedAPI                         ResponseCode = 43
-	UnexpectedError                        ResponseCode = 99
+	Success                                ResponseCode = 2000
```

Comment:
These are values that may already be in use by CNS API clients. As arbitrary as they seem, they cannot be changed with any existing CNS API version.

Reply:
I've updated the reason for the change to better explain the purpose of these changes.

Reply:
Welcome @abel2-code, please take a look at https://github.com/Azure/azure-container-networking/blob/master/docs/contributing/github.md when you get the chance. 

For now, lets allow @ramiro-gamarra to close the conversation when he feels it is resolved. ✌️